app/services/hyrax/workflow/deposited_notification.rb

```diff
@@ -9,13 +9,17 @@ def subject
       end
 
       def message
-        I18n.t('hyrax.notifications.workflow.deposited.message', title: title, link: (link_to work_id, document_path),
-                                                                 user: user.user_key, comment: comment)
+        I18n.t('hyrax.notifications.workflow.deposited.message',
+               title: title,
+               link: (link_to work_id, document_path),
+               user: user.user_key,
+               comment: comment)
       end
 
       def users_to_notify
-        user_key = Hyrax.query_service.find_by_alternate_identifier(alternate_identifier: work_id).depositor
-        super << ::User.find_by(email: user_key)
+        user_key = @entity.proxy_for.depositor
```

Comment:
In the recent PRs for granting/revoking permissions, you checked that depositor method exists before using it.  Would the same apply here?

Example:
```
        return true unless target.try(:depositor)
```

Reply:
i think in those cases we want the grant to be a no-op if there's no depositor. here an error seems better than failing to notify silently.